### PR TITLE
Move setting of AA_EnableHighDpiScaling before creating QApplication.

### PR DESCRIPTION
**Pull request by anntzer**
This is required by the docs
(https://doc.qt.io/qt-5/qt.html#ApplicationAttribute-enum)
and Qt 5.12.4 has started emitting a warning otherwise
("Attribute Qt::AA_EnableHighDpiScaling must be set before
QCoreApplication is created.")

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/backends/backend_qt5.py [hunk 1]
```diff
@@ -118,6 +118,11 @@ def _create_qApp():
                 if display is None or not re.search(r':\d', display):
                     raise RuntimeError('Invalid DISPLAY variable')
 
+            try:
+                QtWidgets.QApplication.setAttribute(
+                    QtCore.Qt.AA_EnableHighDpiScaling)
+            except AttributeError:
+                pass
```
Comment (timhoffm) [inline lib/matplotlib/backends/backend_qt5.py]: Would add a comment `# QtCore.Qt.AA_EnableHighDpiScaling only available on Qt >= 5.6` to clarify why we expect an AttributeError.

Comment (anntzer) [reply lib/matplotlib/backends/backend_qt5.py]: sure.

File: lib/matplotlib/backends/backend_qt5.py [hunk 2]
```diff
@@ -118,6 +118,11 @@ def _create_qApp():
                 if display is None or not re.search(r':\d', display):
                     raise RuntimeError('Invalid DISPLAY variable')
 
+            try:
+                QtWidgets.QApplication.setAttribute(
+                    QtCore.Qt.AA_EnableHighDpiScaling)
```
Comment (dstansby) [inline lib/matplotlib/backends/backend_qt5.py]: This was inside a `if is_pyqt5()` block below; does running this still work fine on Qt4?

Comment (anntzer) [reply lib/matplotlib/backends/backend_qt5.py]: that'll be caught by the AttributeError too.

**Comment by tacaswell** [comment]
We may need to put a similar change into ipython as they frequently create the QApplication instance we use.

**Pull request merged**